File: src/uimd/dialogs/message_box_ok_cancel.py
"""Standard OK/Cancel message box."""

try:
    from uimd.dialogs.message_box import MessageBoxBehavior
    from uimd.dialogs.message_box_ok_cancel_ui import MessageBoxOkCancelUI
except ImportError:
    from message_box import MessageBoxBehavior
    from message_box_ok_cancel_ui import MessageBoxOkCancelUI


class MessageBoxOkCancel(MessageBoxBehavior, MessageBoxOkCancelUI):
    def __init__(self, header="Message", message="", on_close=None):
        super().__init__()
        self._configure_message_box(
            header=header,
            message=message,
            on_close=on_close,
            result_by_button={
                "ok_btn": "ok",
                "cancel_btn": "cancel",
            },
            default_result="cancel",
        )

File: java/src/main/java/uimd/MessageBoxOkCancel.java
package uimd;

import java.util.Map;
import java.util.function.Consumer;

public final class MessageBoxOkCancel extends MessageBoxOkCancelUI
{
    private final MessageBoxBehavior behavior;

    public MessageBoxOkCancel()
    {
        this("Message", "", null);
    }

    public MessageBoxOkCancel(String header, String message)
    {
        this(header, message, null);
    }

    public MessageBoxOkCancel(String header, String message, Consumer<String> onClose)
    {
        behavior = new MessageBoxBehavior(
            dialog_header,
            this.message,
            header,
            message,
            onClose,
            Map.of("ok_btn", "ok", "cancel_btn", "cancel"),
            "cancel");
    }

    public String result()
    {
        return behavior.result();
    }

    public boolean closed()
    {
        return behavior.closed();
    }

    public int run(String[] arguments)
    {
        return GeneratedWindowRuntime.runGeneratedWindow(this, runtimeOptions(), arguments);
    }

    @Override
    public GeneratedWindowRuntimeOptions runtimeOptions()
    {
        return behavior.configure(super.runtimeOptions());
    }

    @Override
    protected void onOkBtnClick()
    {
        behavior.handleButton("ok_btn");
    }

    @Override
    protected void onCancelBtnClick()
    {
        behavior.handleButton("cancel_btn");
    }
}
